reframe/frontend/cli.py

```diff
@@ -183,9 +183,9 @@ def main():
         help='Specify the maximum number of times a failed regression test '
              'may be retried (default: 0)')
     run_options.add_argument(
-        '--flex-alloc-tasks', action='store',
-        dest='flex_alloc_tasks', metavar='{all|idle|NUM}', default='idle',
-        help="Strategy for flexible task allocation (default: 'idle').")
+        '--flex-alloc-nodes', action='store',
+        dest='flex_alloc_nodes', metavar='{all|idle|NUM}', default='idle',
+        help="Strategy for flexible node allocation (default: 'idle').")
```

Comment:
You should deprecate the other option, not delete it. If specified, you should print a deprecation warning and either exit or simply set the `flex_alloc_nodes`.